spec/services/adjustment_create_service_spec.rb

```diff
@@ -0,0 +1,117 @@
+      expect do
+        adjustment_params = {user_id: @user.id, organization_id: @organization.id, storage_location_id: storage_location.id, line_items_attributes: {"0": {item_id: storage_location.items.first.id, quantity: -5}}}
+        subject.new(adjustment_params).call
+      end.to change { Adjustment.count }.by(1)
+
+      adjustment = Adjustment.last
+      expect(adjustment.line_items.count).to eq(1)
+      expect(adjustment.line_items[0].quantity).to eq(-5)
+    end
+
+    it "increases handles mixed adjustments to same item appropriately (total is positive version)" do
+      expect do
+        adjustment_params = {user_id: @user.id,
+                             organization_id: @organization.id,
+                             storage_location_id: storage_location.id,
+                             line_items_attributes: {
+                               "0": {item_id: storage_location.items.first.id, quantity: 4},
+                               "1": {item_id: storage_location.items.first.id, quantity: -5},
+                               "2": {item_id: storage_location.items.first.id, quantity: 2}
+                             }}
+        subject.new(adjustment_params).call
+      end.to change { inventory_item_1.reload.quantity }.by(1)
+      adjustment = Adjustment.last
+      expect(adjustment.line_items.count).to eq(1)
+      expect(adjustment.line_items[0].quantity).to eq(1)
+    end
+
+    it "increases handles mixed adjustments to same appropriately (total is negative version)" do
+      expect do
+        adjustment_params = {user_id: @user.id,
+                             organization_id: @organization.id,
+                             storage_location_id: storage_location.id,
+                             line_items_attributes: {
+                               "0": {item_id: item_1.id, quantity: -4},
+                               "1": {item_id: item_1.id, quantity: -5},
+                               "2": {item_id: item_1.id, quantity: 2}
+                             }}
+        subject.new(adjustment_params).call
+      end.to change { inventory_item_1.reload.quantity }.by(-7)
+      adjustment = Adjustment.last
+      expect(adjustment.line_items.count).to eq(1)
+      expect(adjustment.line_items[0].quantity).to eq(-7)
+    end
+
+    it "does not allow inventory to be adjusted below 0" do
+      quantity = -1 * (inventory_item_1.quantity + 1)
+      expect do
+        adjustment_params = {user_id: @user.id,
+                             organization_id: @organization.id,
+                             storage_location_id: storage_location.id,
+                             line_items_attributes: {
+                               "0": {item_id: item_1.id, quantity: quantity}
+                             }}
+        subject.new(adjustment_params).call
+      end.to change { inventory_item_1.reload.quantity }.by(0)
```

Comment:
Should this store an error, and if so can we check it here?

Reply:
Yes.  It'll take me a few minutes to figure out how to check that said error is thrown